File: 1614-maximum-nesting-depth-of-the-parentheses/1614-maximum-nesting-depth-of-the-parentheses.cpp
class Solution {
public:
    int maxDepth(string s) {
        stack<char>st;
        int maxs=0;
        for(char&ch:s){
            if(ch=='('){
                st.push(ch);
            }
            else if(ch==')'){
                st.pop();
            }
            else{
                continue;
            }
            maxs=max(maxs,(int)(st.size()));
        }
        return maxs;
        
    }
};

File: 1614-maximum-nesting-depth-of-the-parentheses/1614-maximum-nesting-depth-of-the-parentheses.py
class Solution:
    def maxDepth(self, s: str) -> int:
        depth,o=0,0
        for i in range(len(s)):
            ch = s[i]
            if(ch=='('):
                o=o+1
            if(ch==')'):
                o=o-1
            depth=max(depth,o)
        return depth
        